cat-code-arduino/CR-BW0-B00-M00/CR-BW0-B00-M00.ino

```diff
@@ -0,0 +1,371 @@
+  }
+}
+
+
+String ReadEEPROMSettings()
+{
+  if(EEPROM.read(0x00) != SETTINGS_VERSION) {
+    return "";
+  }
+  
+  int newStrLen = EEPROM.read(1);
+  char data[newStrLen + 1];
+  for (int i = 0; i < newStrLen; i++)
+  {
+    data[i] = EEPROM.read(2 + i);
+  }
+  data[newStrLen] = '\0';
+  return data;
+}
+
+void writeSettingsToEEPROM(String settings)
+{
+  byte len = settings.length();
+  EEPROM.write(0, SETTINGS_VERSION); // Magic byte, if settings format changes change this.
+  EEPROM.write(1, len);
+  for (int i = 0; i < len; i++)
+  {
+    EEPROM.write(2 + i, settings[i]);
+  }
+}
+
+void parseSettings(String settings) {
+  int inBL = settings.length();
+  int b = 0;
+  int a = 0;
+  byte eIndex = 0;
+  for(b; b < inBL; b++) {  
+    if (settings[b] == delimiter){
+      eventSet[eIndex] = settings.substring(a,b);
+      a = b+1;
+      eIndex++;                    
+      }
+   }
+}
+
+void ButtonRun(){  
+  for (o=0; o<outputPins_count; o++){                  //looping through Outputpins and setting one at a time to LOW 
+      digitalWrite(outputPins[o],LOW);   
+      delayMicroseconds(0);
+      
+      for(i=0; i<inputPins_count; i++){                // looping through Inputpins and checking for the LOW
+
+          eventIndex = buttonMap[o][i];
+          
+          if(digitalRead(inputPins[i]) == LOW){
+            pressEventSet(eventIndex);
+          }      
+          else {
+            releaseEventSet(eventIndex);      
+          }
+      }      
+    digitalWrite(outputPins[o],HIGH);              //setting the Outputpin back to HIGH 
+//    delayMicroseconds(500);
+    MouseRun();                       // ADNS5050 sub funktion
+    EncoderRun();                         // encoder sub funktion
+  }
+}
+void pressEventSet(byte index){
+  
+  if (EventRunning[index] == false){
+    
+    transientEvent = eventSet[index];
+    eL = transientEvent.length();
+    
+    for(k=0; k < eL; k++){ 
+
+        eventComponent = transientEvent[k];
+      
+        if(eventComponent == mouseClickLeft){
+          Serial.println("LEFT MOUSE PRESSED");            
+            Mouse.press(MOUSE_LEFT);
+        }
+        else if(eventComponent == mouseClickMiddle){
+            Mouse.press(MOUSE_MIDDLE);
+        }
+        else if(eventComponent == mouseClickRight){
+            Mouse.press(MOUSE_RIGHT);
+        }
+        else{
+            Keyboard.press(eventComponent);
+            if(eventComponent == ctrl){
+              ctrl_pressed = index;
+            }
+            if(eventComponent == shift){
+              shift_pressed = index;           
+            }              
+        }             
+    }    
+    //printButtonEvent(index);           // function for serial print of buttons
+    EventRunning[index] = true;
+  }
+  else{
+    //do nothing
+    //printButtonEvent(index);  // function for serial print of buttons
+  }
+}
+void releaseEventSet(byte index){
+  if (EventRunning[index] == true){
```

Comment:
I added in a check here, otherwise I was getting issues assigning mouse click to 2 different buttons. It would only ever act as a click and not a drag.